app/models/application.rb

```diff
@@ -66,16 +66,22 @@ class Application < ApplicationRecord
     withdrawn: "withdrawn",
   }
 
+  def funding_eligibility(with_funded_place:)
+    return eligible_for_funding unless with_funded_place
+
+    eligible_for_funding && (funded_place.nil? || funded_place)
+  end
+
   # `eligible_for_dfe_funding?`  takes into consideration what we know
   # about user eligibility plus if it has been previously funded. We need
   # to keep this method in place to keep consistency during the split between
   # ECF and NPQ. In the mid term we will perform this calculation on NPQ and
   # store the value in the `eligible_for_funding` attribute.
-  def eligible_for_dfe_funding?
+  def eligible_for_dfe_funding?(with_funded_place: false)
```

Comment:
is there a way we can avoid bringing in the complexity from ECF here, do we need those methods to be shared or can they be different, something to think about 🤔